### ISSUE TITLE
Process instance is banned after modification fails due to bad FEEL expression

### ISSUE THREAD

**Issue by koevskinikola** [camunda/zeebe#16005]
**Describe the bug**

A process instance with an incorrect FEEL expression is banned when a process instance modification is attempted.

**To Reproduce**

/

**Expected behavior**

The process instance isn't banned if a modification fails due to a bad FEEL expression.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
java.lang.IllegalStateException: Failed to build variable mapping expression: failed to parse expression '{apiKey:"<REDACTED>",from:{name:"<REDACTED>",email:"<REDACTED>"},to:{name:""receiver"",email:input_item},unMappedFieldNotUseInModel:{mailType:"byTemplate"},template:{id:"<REDACTED>",data:{
"name":variables.customer_name,
"notification_mode":"Escilate",
"mode":"Escalation"
}}}': Expected (start-of-input | ifOp | forOp | quantifiedOp | disjunction):1:1, found "{apiKey:\"S"
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.VariableMappingTransformer.parseExpression(VariableMappingTransformer.java:196) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.VariableMappingTransformer.transformInputMappings(VariableMappingTransformer.java:84) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.lambda$transformIoMappings$1(FlowNodeTransformer.java:64) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at java.base/java.util.Optional.map(Unknown Source) ~[?:?]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transformIoMappings(FlowNodeTransformer.java:62) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transform(FlowNodeTransformer.java:38) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformer.FlowNodeTransformer.transform(FlowNodeTransformer.java:21) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformation.TransformationVisitor.visit(TransformationVisitor.java:42) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.model.bpmn.traversal.TypeHierarchyVisitor.visit(TypeHierarchyVisitor.java:39) ~[zeebe-bpmn-model-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.model.bpmn.traversal.ModelWalker.walk(ModelWalker.java:73) ~[zeebe-bpmn-model-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.deployment.model.transformation.BpmnTransformer.transformDefinitions(BpmnTransformer.java:118) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.deployment.DbProcessState.updateInMemoryState(DbProcessState.java:234) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.deployment.DbProcessState.lookupProcessByIdAndPersistedVersion(DbProcessState.java:400) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.state.deployment.DbProcessState.getLatestProcessVersionByProcessId(DbProcessState.java:292) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.behavior.BpmnStateBehavior.getLatestProcessVersion(BpmnStateBehavior.java:123) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.container.CallActivityProcessor.getProcessForProcessId(CallActivityProcessor.java:187) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.container.CallActivityProcessor.lambda$onActivate$1(CallActivityProcessor.java:65) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.util.Either$Right.flatMap(Either.java:366) ~[zeebe-util-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.container.CallActivityProcessor.onActivate(CallActivityProcessor.java:65) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.container.CallActivityProcessor.onActivate(CallActivityProcessor.java:29) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.lambda$processEvent$2(BpmnStreamProcessor.java:144) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.util.Either$Right.ifRightOrLeft(Either.java:381) ~[zeebe-util-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processEvent(BpmnStreamProcessor.java:143) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.lambda$processRecord$0(BpmnStreamProcessor.java:92) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.util.Either$Right.ifRightOrLeft(Either.java:381) ~[zeebe-util-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.processing.bpmn.BpmnStreamProcessor.processRecord(BpmnStreamProcessor.java:89) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.engine.Engine.process(Engine.java:127) ~[zeebe-workflow-engine-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.batchProcessing(ProcessingStateMachine.java:360) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.lambda$processCommand$2(ProcessingStateMachine.java:275) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.db.impl.rocksdb.transaction.ZeebeTransaction.run(ZeebeTransaction.java:98) ~[zeebe-db-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.processCommand(ProcessingStateMachine.java:275) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.tryToReadNextRecord(ProcessingStateMachine.java:233) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.stream.impl.ProcessingStateMachine.readNextRecord(ProcessingStateMachine.java:209) ~[zeebe-stream-platform-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorJob.invoke(ActorJob.java:94) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorJob.execute(ActorJob.java:47) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorTask.execute(ActorTask.java:120) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.executeCurrentTask(ActorThread.java:130) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.doWork(ActorThread.java:108) [zeebe-scheduler-8.4.0.jar:8.4.0]
	at io.camunda.zeebe.scheduler.ActorThread.run(ActorThread.java:227) [zeebe-scheduler-8.4.0.jar:8.4.0]
"
message: "Expected to process record 'TypedRecordImpl{metadata=RecordMetadata{recordType=COMMAND, valueType=PROCESS_INSTANCE_MODIFICATION, intent=MODIFY, authorization=JWT.<REDACTED>.}, value={"processInstanceKey":4503599656175451,"terminateInstructions":[{"elementInstanceKey":4503599656175615}],"activateInstructions":[{"elementId":"Activity_08e8dgb","ancestorScopeKey":-1,"variableInstructions":[],"ancestorScopeKeys":[]}],"activatedElementInstanceKeys":[]}}' without errors, but exception occurred with message 'Failed to build variable mapping expression: failed to parse expression '{apiKey:"<REDACTED>",from:{name:"<REDACTED>",email:"<REDACTED>"},to:{name:""receiver"",email:input_item},unMappedFieldNotUseInModel:{mailType:"byTemplate"},template:{id:"<REDACTED>",data:{
"name":variables.customer_name,
"notification_mode":"Escilate",
"mode":"Escalation"
}}}': Expected (start-of-input | ifOp | forOp | quantifiedOp | disjunction):1:1, found "{apiKey:\"S"'.
```

</p>
</details>

**Environment:**
- OS:  Camunda SaaS
- Zeebe Version:  `8.4.0`
- Configuration: 
- [GCP logs](https://console.cloud.google.com/logs/query;query=resource.type%3D%22k8s_container%22%0Aresource.labels.project_id%3D%22camunda-cloud-240911%22%0Aresource.labels.location%3D%22europe-west1%22%0Aresource.labels.cluster_name%3D%22production-worker-1%22%0Aresource.labels.namespace_name%3D%222f4f578f-d7f2-4034-9454-2f2c518ab11f-zeebe%22%0Aresource.labels.pod_name:%22zeebe-%22%20severity%3E%3DDEFAULT;pinnedLogId=2024-01-18T11:33:26.338941753Z%2Fxu5xvqc4avdyf001;cursorTimestamp=2024-01-18T11:35:01.406727087Z;startTime=2024-01-18T11:32:20.000Z;endTime=2024-01-18T11:35:05.000Z?project=camunda-cloud-240911)
- [JSON logs](https://drive.google.com/file/d/101TH2fGwzWXupj2Uz09M7IUsre4KnDqv/view?usp=sharing)
- [SUPPORT case](https://jira.camunda.com/browse/SUPPORT-20145)

**Comment by korthout** [comment]
Please consider #16243 before making changes to the BPMN tranformers

**Comment by korthout** [comment]
@nicpuppa Can you have a look whether this is similar to the issue that you recently resolved?

**Comment by nicpuppa** [comment]
I had a look at this bug, and I can confirm it is already resolved by #16065 . Since the fix is already available in `8.4.1` and the related support case is already resolved, I'm going to close this issue. 🚀
